### ISSUE TITLE
Client wont start on ubuntu 22.04

### ISSUE THREAD

**Issue by juancirino23** [emqx/MQTTX#1276]
Just did a clean install of mqttx via snap and I get this error
(mqttx:4133837): Gtk-WARNING **: 14:16:29.157: Theme parsing error: gtk.css:1413:23: 'font-feature-settings' is not a valid property name

(mqttx:4133837): Gtk-WARNING **: 14:16:29.160: Theme parsing error: gtk.css:3286:25: 'font-feature-settings' is not a valid property name

(mqttx:4133837): Gtk-WARNING **: 14:16:29.161: Theme parsing error: gtk.css:3748:23: 'font-feature-settings' is not a valid property name
/usr/share/libdrm/amdgpu.ids: No such file or directory
(node:4133837) UnhandledPromiseRejectionWarning: DriverPackageNotInstalledError: SQLite package has not been found installed. Try to install it: npm install sqlite3 --save
    at new DriverPackageNotInstalledError (/snap/mqttx/31/resources/app.asar/node_modules/typeorm/error/DriverPackageNotInstalledError.js:11:28)
    at SqliteDriver.loadDependencies (/snap/mqttx/31/resources/app.asar/node_modules/typeorm/driver/sqlite/SqliteDriver.js:128:19)
    at new SqliteDriver (/snap/mqttx/31/resources/app.asar/node_modules/typeorm/driver/sqlite/SqliteDriver.js:29:15)
    at DriverFactory.create (/snap/mqttx/31/resources/app.asar/node_modules/typeorm/driver/DriverFactory.js:45:24)
    at new Connection (/snap/mqttx/31/resources/app.asar/node_modules/typeorm/connection/Connection.js:54:59)
    at ConnectionManager.create (/snap/mqttx/31/resources/app.asar/node_modules/typeorm/connection/ConnectionManager.js:57:26)
    at /snap/mqttx/31/resources/app.asar/node_modules/typeorm/index.js:219:66
    at step (/snap/mqttx/31/resources/app.asar/node_modules/typeorm/node_modules/tslib/tslib.js:143:27)
    at Object.next (/snap/mqttx/31/resources/app.asar/node_modules/typeorm/node_modules/tslib/tslib.js:124:57)
    at /snap/mqttx/31/resources/app.asar/node_modules/typeorm/node_modules/tslib/tslib.js:117:75
(Use `mqttx --trace-warnings ...` to show where the warning was created)
(node:4133837) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:4133837) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by juancirino23** [comment]
Hi, thanks for the response just tryed it and i get
error: cannot revert "mqttx": no revision to revert to

**Comment by ZoltanKoczan** [comment]
I just checked, but unfortunately, they do not provide revisions or earlier versions via Snap (and I didn't find it elsewhere either).
The 1.9.2 has been released on 28th April, hopefully, they will fix it soon.

**Comment by ysfscream** [comment]
Hi all. Thanks for your report on this issue. Now I can't figure out the exact problem, either. Is your CPU amd64? Judging from the errors, it seems to be a problem with SQLite, but it should be the same as the previous version. Can you try `npm install sqlite3 --save` and see if that fixes it?

I will continue to investigate this issue.

**Comment by ysfscream** [comment]
I have reuploaded the snap package with version 1.9.2-1. I'm unsure if it's because of a problem with the upload command. 

Could everyone please try `sudo snap install mqttx` again and see if it works?

**Comment by EddieTheHead** [comment]
I'm experiencing the same issue, `npm install sqlite3 --save` doesn't help

```~$ mqttx 

(mqttx:450172): Gtk-WARNING **: 14:38:17.514: Theme parsing error: gtk.css:1422:23: 'font-feature-settings' is not a valid property name

(mqttx:450172): Gtk-WARNING **: 14:38:17.516: Theme parsing error: gtk.css:3308:25: 'font-feature-settings' is not a valid property name

(mqttx:450172): Gtk-WARNING **: 14:38:17.516: Theme parsing error: gtk.css:3770:23: 'font-feature-settings' is not a valid property name
(node:450172) UnhandledPromiseRejectionWarning: DriverPackageNotInstalledError: SQLite package has not been found installed. Try to install it: npm install sqlite3 --save
    at new DriverPackageNotInstalledError (/snap/mqttx/32/resources/app.asar/node_modules/typeorm/error/DriverPackageNotInstalledError.js:11:28)
    at SqliteDriver.loadDependencies (/snap/mqttx/32/resources/app.asar/node_modules/typeorm/driver/sqlite/SqliteDriver.js:128:19)
    at new SqliteDriver (/snap/mqttx/32/resources/app.asar/node_modules/typeorm/driver/sqlite/SqliteDriver.js:29:15)
    at DriverFactory.create (/snap/mqttx/32/resources/app.asar/node_modules/typeorm/driver/DriverFactory.js:45:24)
    at new Connection (/snap/mqttx/32/resources/app.asar/node_modules/typeorm/connection/Connection.js:54:59)
    at ConnectionManager.create (/snap/mqttx/32/resources/app.asar/node_modules/typeorm/connection/ConnectionManager.js:57:26)
    at /snap/mqttx/32/resources/app.asar/node_modules/typeorm/index.js:219:66
    at step (/snap/mqttx/32/resources/app.asar/node_modules/typeorm/node_modules/tslib/tslib.js:143:27)
    at Object.next (/snap/mqttx/32/resources/app.asar/node_modules/typeorm/node_modules/tslib/tslib.js:124:57)
    at /snap/mqttx/32/resources/app.asar/node_modules/typeorm/node_modules/tslib/tslib.js:117:75
(Use `mqttx --trace-warnings ...` to show where the warning was created)
(node:450172) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:450172) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```

**Comment by EddieTheHead** [comment]
Thank you, but it didn't work.

**Comment by ysfscream** [comment]
This is strange! If downloading from Github's Release, can it be used directly? As shown in the screenshot below:

<img width="273" alt="image" src="https://user-images.githubusercontent.com/21299158/236369051-6143876a-bec1-4039-b638-56d0e8308ba8.png">

I apologize that I cannot specifically test it because I do not have an Ubuntu environment.

**Comment by juancirino23** [comment]
you should be able to sudo snap install MQTTX_1.9.2_amd64.snap

Ok so the latest version have a preaty big issue its causing the ubuntu to freeze up. I open the aplication everything is fine but for example once I click connect to a saved server it just gets stuck and basically kill my machine and the same thing happens when I close a connection. This is completly random.

**Comment by ysfscream** [comment]
I appreciate your patience. Before proceeding with the following suggestion, please ensure you have backed up any important data or configurations.

Try clearing the application data to see if it resolves the issue you're experiencing with MQTTX version 1.9.2-2. Here's how you can do that:
1. Close the MQTTX application if it's currently running.
2。 Navigate to the MQTTX application data folder:
  - On Linux: ~/.config/MQTTX
4. Delete the MQTTX.db of the folder.

Restart the MQTTX application and reconfigure your settings.

**Comment by ysfscream** [comment]
On my ARM64-based Ubuntu machine, it works fine. Is your machine using the amd64 architecture?

![image](https://user-images.githubusercontent.com/21299158/236657767-eab7997b-035d-4033-8ed8-f6c89160e68e.png)

**Comment by ysfscream** [comment]
On my ARM64-based Ubuntu machine, it works fine. Is your machine using the amd64 architecture?

![image](https://user-images.githubusercontent.com/21299158/236657767-eab7997b-035d-4033-8ed8-f6c89160e68e.png)

**Comment by juancirino23** [comment]
> I appreciate your patience. Before proceeding with the following suggestion, please ensure you have backed up any important data or configurations.
> 
> Try clearing the application data to see if it resolves the issue you're experiencing with MQTTX version 1.9.2-2. Here's how you can do that:
> 
> 1. Close the MQTTX application if it's currently running.
> 2. Navigate to the MQTTX application data folder:
> 
> * On Linux: ~/.config/MQTTX
> 
> 3. Delete the MQTTX.db of the folder.
> 
> Could you restart the MQTTX application and reconfigure your settings?

ok, just followed your instrucctions. So now the system starts and its works. But like I said the frezzes are completly random.

**Comment by juancirino23** [comment]
> On my ARM64-based Ubuntu machine, it works fine. Is your machine using the amd64 architecture?
> 
> ![image](https://user-images.githubusercontent.com/21299158/236657767-eab7997b-035d-4033-8ed8-f6c89160e68e.png)

My lapton has a Ryzen 5 4500 so I would have to say yes but not really sure. This is an example of what its doing you open the menus and nothing is showing followed by an option to force quit or wait (from ubuntu). And if I connect so a server, then go to any Files or window or help options it frezzes cant disconnect I have to wait and see if I get the force quit screen.
![Screenshot from 2023-05-07 07-34-52](https://user-images.githubusercontent.com/93710242/236672453-d5c9b70f-efe4-4465-8559-5c11dbed77d5.png)
![Screenshot from 2023-05-07 07-35-32](https://user-images.githubusercontent.com/93710242/236672470-214319b3-2473-4adf-af53-0c0b36cce2dd.png)

**Comment by ysfscream** [comment]
Good, I can see that you've successfully opened the application on your machine. Could you please share the correct steps to open it?

Regarding the second issue with the freeze, I suspect that it might be caused by a large number of messages. I noticed that you subscribed to the "#" topic. Although MQTTX has improved its performance, handling a massive amount of messages might still cause lagging and process termination issues.

**Comment by juancirino23** [comment]
Hi I did 
1.) snap remove mqttx
2.) snap install --stable mqttx
3.) I went to my users home directory and did npm install sqlite3 --save
Those are the steps I did hope it helps

Yes the topic is set to # but as you can see in both pics Iam not connected at that moment to any server. I had just opened the aplication. Also the mqtt server is a testing server basically my own.

**Comment by ysfscream** [comment]
Hello, we have fixed the issue, please re-download and try the new version. We apologize for the inconvenience caused.

**Comment by EddieTheHead** [comment]
The MQTTX_1.9.2_amd64.snap file works, thank you for the quick fix. I used `sudo snap install Downloads/MQTTX_1.9.2_amd64.snap --dangerous`to install it

**Comment by juancirino23** [comment]
Ok, just completly removed mqttx, download from the site the lates for ubuntu it runs but here are some errors that are causing the application to crash.
![Screenshot from 2023-05-09 13-01-54](https://github.com/emqx/MQTTX/assets/93710242/5a8af509-2eaf-46c2-9bf8-746cbadd9e57)
![Screenshot from 2023-05-09 13-01-37](https://github.com/emqx/MQTTX/assets/93710242/485369ed-29cf-4ffe-ae78-7398fcbede8d)
![Screenshot from 2023-05-09 13-01-26](https://github.com/emqx/MQTTX/assets/93710242/bc2a5bd1-cf00-4583-a5f1-d4a0dafe33fb)
Basically if you open the aplication window option is loading empty and causes the aplication to crash same with help. will upload some connections to test it.

**Comment by juancirino23** [comment]
Ok also just upload a backup I exported of connection. the aplication froze completly dragin the system with it. had to do hard reset of machine.

**Comment by ysfscream** [comment]
@lnksz Hi. Sorry about that. We apologize that we have not fixed the issue with the snap store. You need to download the snap file and install it manually:
```shell
sudo snap install Downloads/MQTTX_1.9.2_amd64.snap --dangerous 
```
We will update version 1.9.3 to the snap store as soon as possible to fix this issue and I will reopen this issue until this problem with snap store is resolved.

**Comment by ysfscream** [comment]
@juancirino23 Again, we apologize. After opening, clicking the top menu and uploading backup files will freeze, right? Can you roll back to version 1.9.1 to test? Version 1.9.2 resulted in some unexpected issues due to the upgrade of internal dependencies. I need to compare with 1.9.1 to see what problems.

It freezes when uploading files, possibly due to the large amount of data. We have not optimized this yet. We will fix and release v1.9.3 as soon as possible.

**Comment by lnksz** [comment]
Hey @ysfscream thanks for the speedy support!

**Comment by juancirino23** [comment]
@ysfcream thanks for the support.
I cant roll back to 1.9.1 since its not available for download in the snap market or on the site.
I will wait for version 1.9.3.

**Comment by ysfscream** [comment]
> What is the schedule for 1.9.3 release? My 1.9.1 version got updated twice, so I cannot revert it back to that version.

@ZoltanKoczan  Hi, maybe this Friday. Now you can download 1.9.2.snap manually on the mqttx.app website.

**Comment by ysfscream** [comment]
Thanks for your patience. The version https://github.com/emqx/MQTTX/releases/tag/v1.9.3 has been released. However, I'm still unsure about the availability through `snap install mqttx` On Ubuntu arm64, I can successfully run the 1.9.3 `AppImage` and `deb` files.

**Comment by juancirino23** [comment]
Hi ok so I just installed version 1.9.3 downloaded from mqttx.app.
The new version installed fined and boots when executed, tested the conecction to a few servers and connected just fine.
The only thing I noticed so far is that the drop downs files, windows,help when clicked on are empty and those three menus cause the app to become unstable.

**Comment by ysfscream** [comment]
OK, thanks @juancirino23  I will close this issue then. I have not been able to reliably reproduce the issue by clicking Files, Windows, and Help. You can open a new issue providing detailed steps to reproduce, and we will work on resolving it entirely.

**Comment by ZoltanKoczan** [comment]
It works for me as well! Thank you very much for the help and for the support!

Keep up the good work! :)
